generator/src/lib.rs

```diff
@@ -2709,6 +2784,7 @@ pub fn generate_feature<'a>(
             &HashMap::new(),
             fn_cache,
             has_lifetimes,
+            "Raw Vulkan 1 static function pointers",
```

Comment:
Do you think it makes sense to start looking into how to generate the same for `Entry`/`Instance`/`Device`? They share an even more complicated ambiguity with `ash::Device`, `ash::DeviceFn1_X` and `ash::vk::Device` (we have open issues/PRs about it). The version separation might make this even more complicated.

Reply:
I'm not sure that makes sense. `ash::vk::{Instance, Device}` are a very different kind of thing than `ash::Device`; it's not like the extensions where we had a function pointer table floating around in complete isolation.

Reply:
Yeah, exactly. And that's my only gripe with the approach we're going for here: the split between `ash::` and `ash::vk::` used to be that the latter contains generated "low-level" `sys`-like bindings, and all the high-level stuff remains in `ash::{Entry,Instance,Device, extensions}`.

Reply:
Specifically, while `vk::Device` exists and is important to preserve as a useful thing, extensions don't generally introduce new dispatchable handles.

I agree that this blurs the boundary a bit and I don't love that, but I think the reduction in confusion from not having duplicate extension module hierarchies is hard to ignore.

What if we go the other way? Flatten `vk::*` back down, but expose the function-pointer tables, `NAME` constants, and so forth exclusively under `ash::extensions`? I think this recovers the graceful separation while still preserving the wins here.

Reply:
> What if we go the other way? Flatten `vk::*` back down, but expose the function-pointer tables, `NAME` constants, and so forth exclusively under `ash::extensions`? I think this recovers the graceful separation while still preserving the wins here.

Should we merge this in, and toy around with that as a followup PR - or push it straight here? I think you're on to another good idea. `ash` in Rust "means" Vulkan bindings, so it doesn't mean much to have an other `vk` module level when the "previously made argument about a `sys` boundary" no longer really applies thanks to this change.

The only problem: how do we address the subsequent naming clash between `ash::Device` and `ash::vk::Device`? Besides renaming (to what?), any extra module hierarchy we might introduce for that (i.e. a theoretical but unlikable `mut wrappers`) introduces the same split we have for extensions before this PR.

Reply:
Added a new commit that revises the module layout per our discussion this morning.